Task3/src/Utils/Utils.hs

```diff
@@ -1,9 +1,19 @@
-module Utils.Utils (maybeHead, unwrap) where
+module Utils.Utils (maybeHead, unwrap, validateArr, validateMaybe) where
 import Data.Maybe (fromMaybe)
+import Data.App
+import Control.Monad.Except
 
 maybeHead :: [a] -> Maybe a
 maybeHead [] = Nothing
 maybeHead (x:_) = Just x
 
-unwrap :: IO (Maybe (IO (Maybe a))) -> IO (Maybe a)
-unwrap m = m >>= fromMaybe (return Nothing)
\ No newline at end of file
+unwrap :: App (Maybe (App (Maybe a))) -> App (Maybe a)
+unwrap m = m >>= fromMaybe (return Nothing)
+
+validateArr :: [a] -> App a
+validateArr [] = throwError ElementNotFound
```

Comment:
Let's use a more descriptive error message: "Customer with id 6 can't be found" or smth like that